ScienceJournal/Sensors/MagnetometerSensor.swift

```diff
@@ -61,15 +62,45 @@ class MagnetometerSensor: MotionSensor {
   }
 
   override func callListenerBlocksWithData(atMilliseconds milliseconds: Int64) {
-    guard let magneticField =
-        MagnetometerSensor.motionManager.deviceMotion?.magneticField.field else { return }
+    guard let magneticField = MagnetometerSensor.motionManager.deviceMotion?.magneticField.field,
+      isValid(magneticFieldReading: magneticField)
+    else { return }
 
     // The magnetic strength is the square root of the sum of the squares of the values in X, Y
     // and Z.
-    let magneticStrength =
+    var magneticStrength =
         sqrt(pow(magneticField.x, 2) + pow(magneticField.y, 2) + pow(magneticField.z, 2))
+
+    // Typical white dwarf stars have a magnetic field of around 100 Tesla (100,000,000 µT), so
+    // let's cap our values there.
+    // Note: Not a good idea to get close enough to measure a white dwarf star.
+    if magneticStrength > whiteDwarfMagneticFieldStrength {
+      magneticStrength = whiteDwarfMagneticFieldStrength
+    }
+
     let dataPoint = DataPoint(x: milliseconds, y: magneticStrength)
     callListenerBlocksWithDataPoint(dataPoint)
   }
 
+  /// Sometimes we can get junk readings. The easiest way to check is to compare all axes
+  /// to make sure their individual readings aren't extremely far apart.
+  private func isValid(magneticFieldReading field: CMMagneticField) -> Bool {
+    // Axes can be negative.
+    let x = abs(field.x)
+    let y = abs(field.y)
+    let z = abs(field.z)
+
+    let largestAxisValue = max(max(x, y), y)
+    let smallestAxisValue = min(min(x, y), z)
+
+    // If we can multiply the smallest value by 10,000 and it's larger than the largest value, that means
```

Comment:
Longer than 100?

Reply:
Yeah, greater than 100 µT

Reply:
Haha, I mean is this line longer than 100 characters :) (And it is, so please break to next line.)